source/routing/routing.h

```diff
@@ -112,6 +112,52 @@ struct PathItem {
     }
 };
 
+inline std::ostream& operator<<(std::ostream& ss, const PathItem& t) {
```

Comment:
Why do you need to inline this ?! There is no template or apparent reason of moving it to a header.
It was better when this was in the .cpp file :(

Reply:
Good catch ! Fixed in 07005c15df4b9e52f9680128932e65490fbdbad4
I think I had some cyclic dependency linking problem at some point, that's why it ended up in the header. Seems to work now.

Reply:
Hum, doesn't compile after a make clean, so the dependecy problem is stil here

Reply:
Ok, fixed in 0e9da87ca6a4d79b03309c273c87e7785c3acb69